packages/eslint-plugin/docs/rules/ban-types.md

```diff
@@ -0,0 +1,22 @@
+:::danger Deprecated
+
+The old `ban-types` rule encompassed multiple areas of functionality, and so has been split into several rules.
+
+**[`no-restricted-types`](./no-restricted-types.mdx)** is the new rule for banning a configurable list of type names.
+It has no options enabled by default.
```

Comment:
> And is akin to rules like `no-restricted-syntax`, `no-restricted-globals`, `no-restricted-exports`, `no-restricted-imports`, `no-restricted-properties`, `id-denylist`.

Idk if it's worth mentioning these or some of these to help contextualise the rule? (ditto for adding this to the new rule's docs).

Reply:
Good call, I added a few in.